network/src/noise_wrapper/handshake.rs

```diff
@@ -0,0 +1,356 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! The handshake module implements the handshake part of the protocol.
+//! This module also implements additional anti-DoS mitigation,
```

Comment:
I'm sceptical of the effectiveness of DoS mitigation through statefulness on an 'I' protocol (by which the initiator doesn't really need to prove anything but knowledge of the receiver's identity on entry) ... do we want an open issue to let admission control introduce some token delivery in the mid-term future, which would allow us to upgrade this to a KK framework?

Reply:
the DoS mitigation is not about the handshake pattern, but about sending a payload as part of the first message to avoid replays : o

Reply:
Yeah. But that's my point: you can, at modest cost, generate a large number of valid distinct first messages which will mandate an answer.